Improve developer guide

Weltraumschaf opened the pull request.

Weltraumschaf committed:
#24 Remove trailing spaces from developerguide

Signed-off-by: Sven Strittmatter <sven.strittmatter@iteratec.com>

Weltraumschaf committed:
#24 Clean up Markdown of developer guide

- Move the documentation as comments in fenced code blocks
  up as toplevel text for better readability.
- Use code (backtick) markup for stuff used as keyboard input
  and filenames.
- Remove leading $ in bash example because this prevents copy
  paste of the examle code w/o manual editing.
- Make single paragraph text also a single line w/o unnecessary
  linebreaks. Just use an editor with line wrapping.
- Some spell checking.
- Markdown allows only one H1 equivalent top level title

Signed-off-by: Sven Strittmatter <sven.strittmatter@iteratec.com>

Signed-off-by: Sven Strittmatter <sven.strittmatter@iteratec.com>

Weltraumschaf committed:
#24 Fix case of titles

There are some rules around how to case titles in English.
I justed used the default of tcc[1]: Chicago Manual of Style.

1: https://github.com/nickjj/title-case-converter

Signed-off-by: Sven Strittmatter <sven.strittmatter@iteratec.com>

Weltraumschaf committed:
#24 Rework developer guide for operator

- Add a macosOS section with Homebrew.
- Linebreak long commands.
- Change the export: We can just use localhost.
- Add hint about direnv.
- Clarify instructions about uninstalling the operator.

Signed-off-by: Sven Strittmatter <sven.strittmatter@iteratec.com>

Weltraumschaf committed:
#24 Add direnv configs for Minio

This enables a per direcotry configuration which does not
mangle up with other projects using S3.

Signed-off-by: Sven Strittmatter <sven.strittmatter@iteratec.com>

Weltraumschaf committed:
#24 Shorthand script to install everything

It is cumbersome to copy paste the snippets always from the README.

Signed-off-by: Sven Strittmatter <sven.strittmatter@iteratec.com>

nigthknight commented on bin/install-all.sh:
```diff
@@ -0,0 +1,24 @@
+set -eu
+
+kubectl create namespace securecodebox-system
+helm -n securecodebox-system install securecodebox-operator ./operator/
```
I would recommend installing the operator with "helm -n securecodebox-system upgrade --install securecodebox-operator ./operator/" as well

Weltraumschaf replied:
Done

Weltraumschaf committed:
#24 Also upgrade install the operator

For the sake of consistency we also upgrade install the operator
instead of only installing it.

Signed-off-by: Sven Strittmatter <sven.strittmatter@iteratec.com>
